egret/model_library/unit_commitment/uc_utils.py

```diff
@@ -124,3 +126,27 @@ def get_linear_expr(*args):
         if not is_var(arg):
             return linear_summation
     return _linear_expression
+
+def make_penalty_rule(m, penalty_key, divisor):
+    def penalty_rule(m):
+        return m.model_data.data['system'].get(penalty_key, m.LoadMismatchPenalty/divisor)
```

Comment:
Don't you want a pyo.value() around the m.LoadMismatchPenalty/divisor logic? I don't think you want an expression...

Reply:
Although given that Param presumably wraps the supplied value with pyo.value(), it's probably fine. Just looks weird, given the penalty-key route obviously gives a float instead of an expression.

Reply:
Yes that's right. I originally was trying to do something cuter, but it didn't work out.